evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -568,6 +612,108 @@ impl<'a, 'b, B: Backend, P: PrecompileSet> CheatcodeStackExecutor<'a, 'b, B, P>
         }
     }
 
+    /// Debug execute a transaction by grabbing the debugger from state_mut, adding the runtime.
+    pub fn debug_execute(
+        &mut self,
+        runtime: &mut Runtime,
+        address: Address,
+        code: Rc<Vec<u8>>,
+    ) -> ExitReason {
+        let mut forge_runtime = ForgeRuntime::new_with_runtime(runtime, code);
+        let depth = if let Some(depth) = self.state().metadata().depth() { depth + 1 } else { 0 };
+
+        match self.debug_run(&mut forge_runtime, address, depth) {
+            Capture::Exit(s) => s,
+            Capture::Trap(_) => unreachable!("Trap is Infallible"),
+        }
+    }
+
+    pub fn debug_step(&mut self, runtime: &ForgeRuntime, steps: &mut Vec<DebugStep>) -> bool {
+        let pc = if let Ok(pos) = runtime.inner.machine().position() { *pos } else { 0 };
+        let mut push_bytes = None;
+        if let Some((op, stack)) = runtime.inner.machine().inspect() {
+            let wrapped_op = OpCode(op, None);
+            if let Some(push_size) = wrapped_op.push_size() {
+                let push_start = pc + 1;
+                let push_end = pc + 1 + push_size as usize;
+                if push_end < runtime.code.len() {
+                    push_bytes = Some(runtime.code[push_start..push_end].to_vec());
+                } else {
+                    panic!("PUSH{} exceeds codesize?", push_size)
+                }
+            }
+            let mut stack = stack.data().clone();
+            stack.reverse();
```

Comment:
because sputnik's *oldest* stack object is at the front and newest at the back. To have it formatted correctly in the tui, i just reverse it here. Open to moving this reversal